### PR TITLE
[AGENT] Reduce TaggedFlow memory consumption

### PR DESCRIPTION
**Pull request by rvql**
### This PR is for:
- Agent

### Improves the performance of FlowMap
Reduce memory consumption from 2.88GB to 2.16GB when new flow rate is 12.8K/s

image

### CODE REVIEW

File: agent/src/collector/l7_quadruple_generator.rs [hunk 1]
```diff
@@ -257,17 +262,25 @@ impl SubQuadGen {
 
             // If l7_stats.flow.is_some(), set the flow of all meter belonging to this flow
             if let Some(tagged_flow) = &l7_stats.flow {
+                let mut flow = MiniFlow::from(&tagged_flow.flow);
+                // TODO: Move direction into tagged_flow to avoid redundant `get_direction` call
```
Comment (TomatoMr) [inline agent/src/collector/l7_quadruple_generator.rs:266]: 要不直接在 tagged_flow.flow 中加个 directions: [Direction, 2] 字段，在 flow_map 中 init_flow 的 set_tap_side 时可以设置 directions

Comment (rvql) [reply agent/src/collector/l7_quadruple_generator.rs:266]: 可以，最好是 directions 和 tap_side 只保留一个以避免不一致

Comment (rvql) [reply agent/src/collector/l7_quadruple_generator.rs:266]: @TomatoMr 转了个 issue 给你 https://github.com/deepflowio/deepflow/issues/4401

Comment (TomatoMr) [reply agent/src/collector/l7_quadruple_generator.rs:266]: 好的

**Review by sharang — APPROVED** [review]

**Pull request merged**